plugins/modules/record.py

```diff
@@ -81,7 +81,7 @@ def main():
     purge = module.params.get("purge")
 
     if zone_id is None:
-        zones = dns.get_zone_info()
+        zones = dns.get_zone_info(zone_name)
```

Comment:
The only issue I see is, that `zone_name` might be undefined, because of

```python
         required_one_of=[
            ['zone_id', 'zone_name']
        ],
```

but the rest of the code looks like that I never worked when just the `zone_id` is provided :)  
And maybe no one will use it with the `zone_id`.

Reply:
You're right, the zone_name does not have to be defined.

>  And maybe no one will use it with the zone_id.

Yeah, I think so, too. I personally never do.

Should I revert this commit?

Reply:
ah nvm.

Reply:
> You're right, the zone_name does not have to be defined.
> 
> > And maybe no one will use it with the zone_id.
> 
> Yeah, I think so, too. I personally never do.
> 
> Should I revert this commit?

The code section was only executed when `zone_name` was requested.  
`if zone_id is None:`  
So - to my surprise - the module works flawless when the `zone_id` is requested :)

I released your changes with 1.5.0 minutes ago.  

Thx @rndmh3ro